src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,252 @@
+
+            Map<String, String> newParams = newTrip.parseQueryParams();
+            Assertions.assertEquals(date, newParams.get(DATE_PARAM));
+        }
+    }
+
+    @Test
+    public void testGetDatesToCheckItineraryExistence() throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(false);
+        trip.monday = true;
+        trip.tuesday = true;
+        trip.wednesday = false;
+        trip.thursday = true;
+        trip.friday = false;
+        trip.saturday = true;
+        trip.sunday = true;
+
+        // The list includes dates to be monitored in a 7-day window starting from the query date.
+        List<String> newDates = List.of(QUERY_DATE /* Thursday */, "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18");
+        List<String> checkedDates = ItineraryUtils.getDatesToCheckItineraryExistence(trip, false);
+        Assertions.assertEquals(newDates, checkedDates);
+
+        // If we forceAllDays to ItineraryUtils.getDatesToCheckItineraryExistence,
+        // it should return all dates regardless of the ones set in the monitored trip.
+        List<String> allDates = List.of(QUERY_DATE /* Thursday */, "2020-08-14", "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18", "2020-08-19");
+        List<String> allCheckedDates = ItineraryUtils.getDatesToCheckItineraryExistence(trip, true);
+        Assertions.assertEquals(allDates, allCheckedDates);
+    }
+
+    @Test
+    public void testAddIgnoreRealtimeParam() throws URISyntaxException {
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            MonitoredTrip trip = new MonitoredTrip();
+            trip.queryParams = query;
+            Map<String, String> params = ItineraryUtils.excludeRealtime(trip.parseQueryParams());
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws IOException, URISyntaxException {
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = BASE_QUERY + "&ui_activeItinerary=1";
+        trip.itinerary = null;
+
+        String mockResponse = FileUtils.getFileContents(
+            TEST_RESOURCE_PATH + "persistence/planResponse.json"
+        );
+        OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse(mockResponse, URI.create("http://www.example.com"));
+        List<Itinerary> itineraries = otpDispatcherResponse.getResponse().plan.itineraries;
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    private void testIsSameDay(String time, boolean isArrival, boolean expected, Long... startTimes) throws URISyntaxException {
```

Comment:
This method along with all the methods that call it could be combined into a single ParameterizedTest. The test result looks a lot cleaner and if one of the test cases fails, the others will still be ran.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/097c7f2bae3c9e29bcd6894de370cedefdaa3e12.